src/cargo/util/toml/targets.rs

```diff
@@ -536,11 +548,24 @@ fn to_bench_targets(
     Ok(result)
 }
 
+fn is_resolved(toml_targets: Option<&Vec<TomlTarget>>, autodiscover: Option<bool>) -> bool {
+    if autodiscover != Some(false) {
+        return false;
+    }
+
+    let Some(toml_targets) = toml_targets else {
+        return true;
+    };
+    toml_targets
+        .iter()
+        .all(|t| t.name.is_some() && t.path.is_some())
+}
```

Comment:
does this mean that if in the `Cargo.toml` (whether normalized or not) all `[[bin]]` entries contain a `name` and `path` field then the auto-discovery of additional bins is disabled? that seems to have potential for false positives and would not match https://doc.rust-lang.org/cargo/reference/cargo-targets.html#target-auto-discovery. last time this behavior was changed it was done at an edition boundary 🤔

Reply:
We only get here if autodiscover is explicitly disabled. We reuse autodiscover logis is reused to fill in `name` or `path` and this is checking to see if that is needed.

If there is a behavior change you noticed, could you be more explicit about what it is?